src/controller.js

```diff
@@ -0,0 +1,104 @@
+(function exportController() {
+  class Controller {
+    constructor(ship) {
+      this.ship = ship;
+      this.initialiseSea();
+      document.querySelector('#setSailButton').addEventListener('click', () => this.setSail());
+    }
+
+    initialiseSea() {
+      const backgrounds = ['./images/water0.png', './images/water1.png'];
+      let backgroundIndex = 0;
+
+      window.setInterval(() => {
+        document.querySelector('#viewport').style.backgroundImage = `url('${backgrounds[backgroundIndex % backgrounds.length]}')`;
+        backgroundIndex += 1;
+      }, 1000);
+    }
+
+    renderPorts(ports) {
+      const portsElement = document.querySelector('#ports');
+      portsElement.style.width = '0px';
+      ports.forEach((port, ind) => {
+        const portDiv = document.createElement('div');
+        portDiv.setAttribute('class', 'port');
+        portDiv.dataset.portIndex = ind;
+        portDiv.dataset.portName = port.name;
+        portsElement.appendChild(portDiv);
+        const portsElWidth = parseInt(portsElement.style.width, 10);
+        portsElement.style.width = `${portsElWidth + 256}px`;
+      });
+    }
+
+    renderShip() {
+      const ship = this.ship;
+      const portIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+      const portEl = document.querySelector(`[data-port-index='${portIndex}']`);
+      const shipElement = document.querySelector('#ship');
+      shipElement.style.top = `${portEl.offsetTop + 32}px`;
+      shipElement.style.left = `${portEl.offsetLeft - 32}px`;
+    }
+
+    renderMessage(message) {
+      const viewport = document.querySelector('#viewport');
+      const messageBox = document.createElement('div');
+      messageBox.setAttribute('id', 'message');
+      messageBox.innerHTML = message;
+      viewport.appendChild(messageBox);
+      window.setTimeout(() => viewport.removeChild(messageBox), 2000);
+    }
+
+    headUpDisplay() {
+      const ship = this.ship;
+      const portIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+      const nextPortIndex = portIndex + 1;
+      const currentHtml = document.getElementById('currentPort');
+      currentHtml.innerHTML = `Current Port: ${this.ship.itinerary.ports[portIndex].name}`;

```

Comment:
`const` declarations should be grouped together, and some spacing between those an the `if` statement after would be great